### DISCUSSION TITLE
Keep "Repeat on interval" running even when moving between request

### DISCUSSION THREAD

**Discussion by wederbrand** [Kong/insomnia#3706] [Ideas]
- Insomnia Version: 6.0.2
- Operating System: Linux

## Details

I need to fetch new authentication cookies periodically during the entire day. I can do that with "Repeat on interval" but that stops as soon as a move to another request. 

I would like it to continue to run forever in the background, feeding new cookies into the cookie jar continuously.

**Comment by gschier** [comment]
This seems like a good idea. The "Repeat on Interval" wasn't a very well designed feature originally and could use some work.

**Comment by nilaymaj** [comment]
I had a go at this, and was able to get it working - the request runs in intervals in the background as needed. I added an `intervalID: IntervalID` property to the request model, since storing interval info in the React component will carry over the interval to the next request too. The component property now mirrors the request model property on mounts and prop changes.

The issue is that if the app is shut down without stopping the interval, on app restart the component has no way to know or check that the interval ID stored in request model is not really an active interval, and thus mirrors a non-functioning interval ID, showing the Stop button.

@gschier Is there some way to null the `intervalID` property of all requests when the app starts (or exits)? Or is there a better implementation?
